### PR TITLE
Add Azure KeyManagementService

### PR DESCRIPTION
**Pull request by jenniew**

### CODE REVIEW

**Comment by jenniew** [comment]
Jenkins passed.

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala [hunk 1]
```diff
@@ -89,6 +91,10 @@ class PPMLContext protected(kms: KeyManagementService, sparkSession: SparkSessio
   def write(dataFrame: DataFrame, cryptoMode: CryptoMode): EncryptedDataFrameWriter = {
     new EncryptedDataFrameWriter(sparkSession, dataFrame, cryptoMode, dataKeyPlainText)
   }
+
+  def getSparkSession(): SparkSession = {
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala:95]: scala doc

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/utils/KeyReaderWriter.scala [hunk 2]
```diff
@@ -16,19 +16,39 @@
...
-    }
+  def writeKeyToFile(encryptedKeyPath: String, encryptedKeyContent: String,
+                     config: Configuration = null): Unit = {
+//    new PrintWriter(encryptedKeyPath) {
+//      write(encryptedKeyContent); close
+//    }
+    val hadoopConfig = if (config != null) config else new Configuration()
+    val fs: FileSystem = FileSystem.get(new URI(encryptedKeyPath), hadoopConfig)
+    val outputStream = fs.create(new Path(encryptedKeyPath))
+    println(s"write key to file: ${encryptedKeyContent}")
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/utils/KeyReaderWriter.scala:37]: remove the `println` if not necessary.

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala (hunk 1 above)
Comment (jenniew) [reply scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala:95]: Will fix in https://github.com/intel-analytics/BigDL/pull/4783

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/utils/KeyReaderWriter.scala (hunk 2 above)
Comment (jenniew) [reply scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/utils/KeyReaderWriter.scala:37]: Will fix in https://github.com/intel-analytics/BigDL/pull/4783

**Pull request merged**